modules/cli/src/main/scala/scala/cli/commands/fix/Fix.scala

```diff
@@ -0,0 +1,363 @@
+
+      val allDirectives = for {
+        transformedMainDirective <- transformedMainDirectives
+        directive                <- transformedMainDirective.directives
+      } yield directive
+
+      val (testScopeDirectives, allMainDirectives) =
+        allDirectives.partition(_.key.startsWith("test"))
+
+      createFormattedLinesAndAppend(allMainDirectives, projectFileContents, isTest = false)
+
+      (
+        transformedMainDirectives.filter(d => isExtractedFromWritableInput(d.positions)),
+        testScopeDirectives
+      )
+    }
+
+    // Deal with directives from the Test scope
+    val directivesFromWritableTestInputs: Seq[TransformedTestDirectives] =
+      if (
+        testSources.paths.nonEmpty || testSources.inMemory.nonEmpty || testDirectivesFromMain.nonEmpty
+      ) {
+        val originalTestDirectives = getExtractedDirectives(testSources)
+          .filterNot(hasTargetDirectives)
+
+        val transformedTestDirectives = unifyCorrespondingNameAliases(originalTestDirectives)
+          .pipe(maybeTransformIntoTestEquivalent)
+
+        val allDirectives = for {
+          directivesWithTestPrefix <- transformedTestDirectives.map(_.withTestPrefix)
+          directive                <- directivesWithTestPrefix ++ testDirectivesFromMain
+        } yield directive
+
+        createFormattedLinesAndAppend(allDirectives, projectFileContents, isTest = true)
+
+        transformedTestDirectives
+          .filter(ttd => isExtractedFromWritableInput(ttd.positions))
+      }
+      else Seq(TransformedTestDirectives(Nil, Nil, None))
+
+    projectFileContents.append(newLine)
+
+    // Write extracted directives to project.scala
+    logger.message(s"Writing ${Constants.projectFileName}")
+    os.write.over(inputs.workspace / Constants.projectFileName, projectFileContents.toString)
+
+    def isProjectFile(position: Option[Position.File]): Boolean =
+      position.exists(_.path.contains(inputs.workspace / Constants.projectFileName))
+
+    // Remove directives from their original files, skip the project.scala file
+    directivesFromWritableMainInputs
+      .filterNot(e => isProjectFile(e.positions))
+      .foreach(d => removeDirectivesFrom(d.positions))
+    directivesFromWritableTestInputs
+      .filterNot(ttd => isProjectFile(ttd.positions))
+      .foreach(ttd => removeDirectivesFrom(ttd.positions, toKeep = ttd.noTestPrefixAvailable))
+  }
+
+  def getProjectSources(inputs: Inputs): Either[::[BuildException], (Sources, Sources)] = {
+    val buildOptions = BuildOptions()
+
+    val (crossSources, _) = CrossSources.forInputs(
+      inputs,
+      preprocessors = Sources.defaultPreprocessors(
+        buildOptions.archiveCache,
+        buildOptions.internal.javaClassNameVersionOpt,
+        () => buildOptions.javaHome().value.javaCommand
+      ),
+      logger = logger,
+      suppressWarningOptions = SuppressWarningOptions(
+        Some(true),
+        Some(true),
+        Some(true)
+      ),
```

Comment:
```suggestion
      suppressWarningOptions = SuppressWarningOptions.suppressAll()
```

If a new field is added to `SuppressWarningOptions` with a default value, this code will have to be updated as well. Perhaps we should create a variable in `SuppressWarningOptions` object that suppresses all warnings